game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProPurchaseTerritory.java

```diff
@@ -64,7 +65,8 @@ public ProPurchaseTerritory(
   public int getRemainingUnitProduction() {
     int remainingUnitProduction = unitProduction;
     for (final ProPlaceTerritory ppt : canPlaceTerritories) {
-      remainingUnitProduction -= ppt.getPlaceUnits().size();
+      remainingUnitProduction -=
+          CollectionUtils.countMatches(ppt.getPlaceUnits(), Matches.unitIsConstruction().negate());
```

Comment:
It's a bit difficult to follow this line. An intermediate variable maybe would help, or adding a `Matches` for 'notConstruction' to avoid the negate. Have you considered unit testing this method to help document and verify it?

Reply:
Updated.